src/magfie/bdivfree.f90

```diff
@@ -181,6 +181,51 @@ subroutine vector_potentials(nr_in,np_in,nz_in,ntor_in,      &
 102 format(1000e15.7)
 end subroutine vector_potentials
 
+
+! Spline a single toroidal Fourier mode of the RMP coil field, as needed for MEPHIT.
+! Note the explicit-shape input arrays are not checked for shape.
+! Also, the truncation of data outside the limiting convex is not performed.
+subroutine vector_potential_single_mode(ntor_in, nR_in, nZ_in, Rmin_in, Rmax_in, Zmin_in, Zmax_in, Bn_R, Bn_Z)
+  use iso_fortran_env, only: dp => real64
+  use bdivfree_mod, only: nR, nZ, Rmin, Zmin, ntor, icp, ipoint, hR, hZ, Rpoi, Zpoi, AZnRe, AZnIm, ARnRe, ARnIm
+  integer, intent(in) :: nR_in, nZ_in
+  real(dp), intent(in) :: Rmin_in, Rmax_in, Zmin_in, Zmax_in
+  complex(dp), intent(in), dimension(nR_in, nZ_in) :: Bn_R, Bn_Z
+  integer :: k, imi(nZ_in), ima(nZ_in), jmi(nR_in), jma(nR_in)
```

Comment:
In know it's nice if the call to `s2dcut` fits one line, but still please use a bit more speaking variable names. I guess it means min and max of something, but what?

Reply:
These are the names used in `vector_potentials` and [`s2dcut`](../tree/main/src/magfie/spline5_RZ.f90), but I'm open to suggestions.